test/common_interface/precs.jl

```diff
@@ -2,92 +2,110 @@ using Sundials, Test, ModelingToolkit, LinearAlgebra, IncompleteLU
+    if jok
+        prob_ode_brusselator_2d_mtk.f.jac(jaccache, u, p, t)
+        jcurPtr[] = true
 
-    # W = I - gamma*J
-    @. W = -gamma*jaccache
-    idxs = diagind(W)
-    @. @view(W[idxs]) = @view(W[idxs]) + 1
+        # W = I - gamma*J
+        @. W = -gamma * jaccache
+        idxs = diagind(W)
+        @. @view(W[idxs]) = @view(W[idxs]) + 1
 
-    # Build preconditioner on W
-    preccache[] = ilu(W, τ = 5.0)
-  end
+        # Build preconditioner on W
+        preccache[] = ilu(W; τ=5.0)
+    end
 end
 
-function precilu(z,r,p,t,y,fy,gamma,delta,lr)
-  ldiv!(z,preccache[],r)
+function precilu(z, r, p, t, y, fy, gamma, delta, lr)
+    ldiv!(z, preccache[], r)
 end
 
-prectmp2 = AlgebraicMultigrid.aspreconditioner(AlgebraicMultigrid.ruge_stuben(W, presmoother = AlgebraicMultigrid.Jacobi(rand(size(W,1))), postsmoother = AlgebraicMultigrid.Jacobi(rand(size(W,1)))))
+prectmp2 = AlgebraicMultigrid.aspreconditioner(AlgebraicMultigrid.ruge_stuben(W;
+                                                                              presmoother=AlgebraicMultigrid.Jacobi(rand(size(W,
+                                                                                                                              1))),
+                                                                              postsmoother=AlgebraicMultigrid.Jacobi(rand(size(W,
+                                                                                                                               1)))))
 const preccache2 = Ref(prectmp2)
 function psetupamg(p, t, u, du, jok, jcurPtr, gamma)
-  if jok
-    prob_ode_brusselator_2d_mtk.f.jac(jaccache,u,p,t)
-    jcurPtr[] = true
+    if jok
+        prob_ode_brusselator_2d_mtk.f.jac(jaccache, u, p, t)
+        jcurPtr[] = true
 
-    # W = I - gamma*J
-    @. W = -gamma*jaccache
-    idxs = diagind(W)
-    @. @view(W[idxs]) = @view(W[idxs]) + 1
+        # W = I - gamma*J
+        @. W = -gamma * jaccache
+        idxs = diagind(W)
+        @. @view(W[idxs]) = @view(W[idxs]) + 1
 
-    # Build preconditioner on W
-    preccache2[] = AlgebraicMultigrid.aspreconditioner(AlgebraicMultigrid.ruge_stuben(W, presmoother = AlgebraicMultigrid.Jacobi(rand(size(W,1))), postsmoother = AlgebraicMultigrid.Jacobi(rand(size(W,1)))))
-  end
+        # Build preconditioner on W
+        preccache2[] = AlgebraicMultigrid.aspreconditioner(AlgebraicMultigrid.ruge_stuben(W;
+                                                                                          presmoother=AlgebraicMultigrid.Jacobi(rand(size(W,
+                                                                                                                                          1))),
+                                                                                          postsmoother=AlgebraicMultigrid.Jacobi(rand(size(W,
+                                                                                                                                           1)))))

```

Comment:
This one looks really weird.

Reply:
yeah that's just an unfortunate that sometimes occurs when using YAS style, if the margin is set to a higher value then it would format nicer that case. `join_lines_based_on_source` still adheres to the margin rules.

Reply:
```julia
        preccache2[] = AlgebraicMultigrid.aspreconditioner(AlgebraicMultigrid.ruge_stuben(W;
                                                                presmoother=AlgebraicMultigrid.Jacobi(rand(size(W,1))),
                                                                postsmoother=AlgebraicMultigrid.Jacobi(rand(size(W,1)))))
```

I don't know what the rule would be, but pulling back often times can make it a bit more legible. Hmm.